### PR TITLE
DOC: add note about localhost & friends in jax.distributed.initialize

### PR DESCRIPTION
**Pull request by nvlcambier**
This PR adds a note in the `jax.distribute.initialize` doc to warn users to not use `localhost` or `127.0.0.1` addresses when doing multi-node runs. This is because using `localhost`, for instance, means that the socket will be bound to a local interface (`lo` for instance). This will prevent other processes to connect to it during bootstrapping and will lead to timeouts (`DEADLINE EXCEEDED`).

### CODE REVIEW

File: jax/_src/distributed.py [hunk 1]
```diff
@@ -131,6 +131,8 @@ def initialize(coordinator_address: Optional[str] = None,
       port does not matter, so long as the port is available on the coordinator
       and all processes agree on the port.
       May be ``None`` only on supported environments, in which case it will be chosen automatically.
+      Note that special addresses like ``localhost`` or ``127.0.0.1`` usually mean that the program
+      will bind to a local interface and are not suited when running in a multi-node environment.
```
Comment (zhangqiaorjc) [inline jax/_src/distributed.py:135]: s/suited/suitable

Comment (yashk2810) [inline jax/_src/distributed.py:135]: Let's call multi-node `multi-process`?

Comment (nvlcambier) [reply jax/_src/distributed.py:135]: Per offline discussion, we'll rename 'multi-hosts'. Multi-process but single-host should work with localhost and/or 127.0.0.1, but I will confirm that.

Comment (nvlcambier) [reply jax/_src/distributed.py]: Fixed

Comment (nvlcambier) [reply jax/_src/distributed.py]: I checked and on a single host, localhost and 127.0.0.1 both work fine.

**Review by zhangqiaorjc — APPROVED** [review]

**Pull request merged**